Initial commit of IRMQTTServer example code.

crankyoldgit opened the pull request:
New code example of how to send IR commands via MQTT and HTTP.
- Support sending most of the simple IR codes. e.g. NEC/Samsung/Sony etc.
- Support sending GlobalCache codes.
- Support sending Kelvinator A/C codes.
- Web form examples for sending codes.
- Url examples for sending codes.
- Instructions in the comments.
- Use WifiManager to set up the network for the device.
- Support OTA firmware updates.

Yes, this is more a real project than just simple example code.

[Tested on a NodeMCU board controlling multiple devices & protocols]

crankyoldgit committed:
Initial commit of the IRMQTTServer example code.
New code example of how to send IR commands via MQTT and HTTP.
- Support sending most of the simple IR codes. e.g. NEC/Samsung/Sony etc.
- Support sending GlobalCache codes.
- Support sending Kelvinator A/C codes.
- Web form examples for sending codes.
- Url examples for sending codes.
- Instructions in the comments.
- Use WifiManager to set up the network for the device.
- Support OTA firmware updates.

Yes, this is more a real project than just example code.

crankyoldgit committed:
Add WifiManager and PubSubClient libraries for IRMQTTServer
- Add to travis
- Add to instructions for mere mortals.

crankyoldgit committed:
HTML clean up and PlatformIO build tweak
- Various minor formatting improvements to the main HTML page.
- Tweak the platformio config to automatically import the required libs.

crankyoldgit committed:
Support repeating codes & fix MQTT parsing bug.
- Add support for sending repeats of simple IR messages.
- Clean up code for default bits.
- [Bug] Fix parsing strings for MQTT non-simple IR messages.
- Add examples of repeats and minor text clean ups.

darshkpatel reviewed (approved):
Would Probably be the most used piece of code

crankyoldgit committed:
Unique mqtt clientid & sort IR types in the form.
- Use a unique clientid for mqtt to avoid future issues.
- Sort the IR protocol types in the form selector.
- Fix typos.
- Tweak the main HTML page formatting.

darshkpatel commented:
Do we merge the PR ?

crankyoldgit commented:
@darshkpatel I was giving it a bit longer for others to comment due to the size of the code review, but it's been a while now. I'll merge.
